catalog/spatial/geocoding/spatial-geocoding-websearch/src/main/java/org/codice/ddf/spatial/geocoder/geonames/GeoNamesWebService.java

```diff
@@ -71,111 +73,80 @@
     private static final String PLACENAME_KEY = "name";
 
     @Override
-    public GeoResult getLocation(String location) {
+    public List<GeoEntry> query(String queryString, int maxResults) throws GeoEntryQueryException {
 
-        location = getUrlEncodedLocation(location);
+        List<GeoEntry> entries = new ArrayList<>();
+        String location = getUrlEncodedLocation(queryString);
 
-        String urlStr = String.format("%s://%s/searchJSON?q=%s&username=%s",
+        String urlStr = String.format("%s://%s/searchJSON?q=%s&username=%s&maxRows=%d",
                 GEONAMES_PROTOCOL,
                 GEONAMES_API_ADDRESS,
                 location,
-                USERNAME);
+                USERNAME,
+                maxResults);
 
-        Object result = query(urlStr);
+        Object result = webQuery(urlStr);
 
         if (result != null) {
             if (result instanceof JSONObject) {
                 JSONObject jsonResult = (JSONObject) result;
-                JSONArray geonames = (JSONArray) jsonResult.get(GEONAMES_KEY);
-                if (geonames != null && geonames.size() > 0) {
-                    JSONObject firstResult = (JSONObject) geonames.get(0);
-                    if (firstResult != null) {
-                        double lat = Double.valueOf((String) firstResult.get(LAT_KEY));
-                        double lon = Double.valueOf((String) firstResult.get(LON_KEY));
-
-                        Long population = (Long) firstResult.get(POPULATION_KEY);
-                        String adminCode = (String) firstResult.get(ADMIN_CODE_KEY);
-
-                        return GeoResultCreator.createGeoResult((String) firstResult.get(
-                                PLACENAME_KEY), lat, lon, adminCode, population);
-                    }
+                JSONArray geoNames = (JSONArray) jsonResult.get(GEONAMES_KEY);
+                if (geoNames != null) {
+                    geoNames.forEach(item -> {
```

Comment:
✏️  Could do this without side effects:
```
return geoNames.stream()
    .map(JSONObject.class::cast)
    .map(obj -> new GeoEntry.Builder()...)
    .collect(toList());
```

and just return `Collections.emptyList()` at the end of the function.